src/lib/crypto/snarky_tests/snarky_tests.ml

```diff
@@ -434,6 +434,44 @@ let api_tests =
   ; ("compile monadic API", `Quick, MonadicAPI.get_hash_of_circuit)
   ]
 
+module Protocol_circuits = struct
+  let proof_level, constraint_constants =
+    Genesis_constants.(Proof_level.Full, Constraint_constants.compiled)
+
+  let blockchain () : unit =
+    let expected = "fe5c73e8189b27aa30ea96076825416e" in
+
+    let digest =
+      Blockchain_snark.Blockchain_snark_state.constraint_system_digests
```

Comment:
so this contains the digests of all the circuits involved in the blockchain snark?

Reply:
looks like it from a glance, then why only check the first one? And why rehash? It sounds like we should check all of them

Reply:
while it is a list, it has a single element, if you look at the function it is a 1 element literal list, I suppose it is that way to compose with some other code that expects a list, will assert length 1 just in case.
What do you mean by rehash? I just take the first and only hash of the list and then convert it to a hex string that I can compare with the literal.

Reply:
I also looked at the transaction test and it has 2 hashes, I added the other one to the test in a way that will fail if the number of hashes changes in the future.